packages/nocodb/src/lib/db/sql-data-mapper/lib/sql/BaseModelSqlv2.ts

```diff
@@ -1839,15 +1849,17 @@ class BaseModelSqlv2 {
 
       await this.beforeUpdate(data, trx, cookie);
 
+      const prevData = await this.readByPk(id);
+
       const query = this.dbDriver(this.tnPath)
         .update(updateObj)
         .where(await this._wherePk(id));
 
       await this.execAndParse(query);
 
-      const response = await this.readByPk(id);
-      await this.afterUpdate(response, trx, cookie);
-      return response;
+      const newData = await this.readByPk(id);
+      await this.afterUpdate(prevData, newData, trx, cookie);
+      return newData;
```

Comment:
Since we have prev data we can insert the row update audit from the backend, at the moment we are calling an extra API call from frontend.

Reply:
Will handle in a separate PR.